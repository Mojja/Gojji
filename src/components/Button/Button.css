.Button {
  background-color: white;
  border: 1px solid black;
  border-radius: 3px;
  padding: 7px 9px;

  .label {
    color: black;
    font-size: 16px;
  }
}
