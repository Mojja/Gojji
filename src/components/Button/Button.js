import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.css'

const Button = ({ onClick, label }) => (
  <button className={styles.Button} onClick={onClick}>
    <img src={require('./x.svg')} alt="x" />
    <span className={styles.label}>{label}</span>
  </button>
)

Button.defaultProps = {
  onClick: null,
}

Button.propTypes = {
  onClick: PropTypes.func,
  label: PropTypes.string.isRequired,
}

export default Button
