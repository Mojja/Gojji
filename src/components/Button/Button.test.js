import React from 'react'
import { create } from 'react-test-renderer'

import Button from './Button'

const state = { isComponentClicked: false }
const _toogleState = () => (state.isComponentClicked = !state.isComponentClicked)

test('Button to be clickable', () => {
  const component = create(<Button label="Awesome button" onClick={_toogleState} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  expect(state.isComponentClicked).toBeTruthy()
})
