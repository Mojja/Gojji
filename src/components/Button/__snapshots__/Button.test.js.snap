// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to be clickable 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  <span
    className="label"
  >
    Awesome button
  </span>
</button>
`;
